Lib/multiprocessing/spawn.py

```diff
@@ -35,6 +35,14 @@
 
 if WINSERVICE:
     _python_exe = os.path.join(sys.exec_prefix, 'python.exe')
+elif (sys.platform == 'win32' and sys.base_exec_prefix != sys.exec_prefix
```

Comment:
For consistency and to avoid checking `sys.platform` twice, how about setting `WINVENV = False` in the existing `sys.platform != 'win32'` block. In the Windows block define `WINVENV = (sys.base_exec_prefix != sys.exec_prefix and '__PYVENV_LAUNCHER__' in os.environ)`.

I presume this means you couldn't resolve this by setting `sys.executable` to the real process image path, maybe because it would require changing too much code. multiprocessing is ours to fix up like this. But other scripts that run `sys.executable` with  a similarly convoluted setup will face the same problem -- i.e. manually duplicating handles into the child via `DuplicateHandle` will dupe them to the launcher process instead.

Reply:
I think there's more likely to be code that modifies environment variables than manually duplicates handles, or uses `sys.executable` later (i.e. writes it into a script), but this is also a much smaller change.

To deal with source tree builds, I have to switch how we generate the new path anyway. So I may as well drop the prefix check because it will essentially work the same in both cases.